src/protocol/vm/engine.rs

```diff
@@ -180,30 +186,25 @@ mod tests {
 
         let engine = create_engine(db, tokens.clone(), false).await;
 
-        // Verify trace flag is unset
-        assert!(!engine.trace);
-
+        let state_data = engine.unwrap().state.data;
         // Verify all tokens are initialized
         for token in tokens {
             let token_address = Address::parse_checksummed(token, None).expect("valid checksum");
-            let account = engine
-                .state
-                .data
+            let account = state_data
+                .clone()
                 .borrow()
                 .get(&token_address)
                 .unwrap()
                 .clone();
```

Comment:
I suspect we don't need to clone this often... we might be able to do something llike:
```suggestion
            let account = state_data.borrow().get(&token_address).cloned().unwrap();
```